docs/standard/threading/overview-of-synchronization-primitives.md

```diff
@@ -1,174 +1,167 @@
+
+## Thread interaction, or signaling
+
+Thread interaction occurs when a thread must wait for notification, or signal, from one or more threads in order to proceed. For example, if a thread A calls the <xref:System.Threading.Thread.Join%2A?displayProperty=nameWithType> method of a thread B, the thread A is blocked until the thread B completes. The synchronization primitives described in the preceding section is another example of signaling: by releasing a lock a thread notifies another thread that it can proceed by acquiring the lock.
+
+This section describes additional signaling constructs provided by .NET.
+
+### EventWaitHandle class
+
+The <xref:System.Threading.EventWaitHandle?displayProperty=nameWithType> class represents a thread synchronization event.
+
+A synchronization event can be either in an unsignaled or signaled state.
+
+When the state of an event is unsignaled, a thread that calls the event's <xref:System.Threading.WaitHandle.WaitOne%2A?> overload is blocked until an event is signaled. The <xref:System.Threading.EventWaitHandle.Set%2A?displayProperty=nameWithType> method sets the state of an event to signaled.
+
+The behavior of an <xref:System.Threading.EventWaitHandle> that has been signaled depends on its reset mode:
+
+- An <xref:System.Threading.EventWaitHandle> created with the <xref:System.Threading.EventResetMode.AutoReset?displayProperty=nameWithType> flag resets automatically after releasing a single waiting thread. It's like a turnstile that allows only one thread through each time it's signaled.
+- An <xref:System.Threading.EventWaitHandle> created with the <xref:System.Threading.EventResetMode.ManualReset?displayProperty=nameWithType> flag remains signaled until its <xref:System.Threading.EventWaitHandle.Reset%2A> method is called. It's like a gate that is closed until signaled and then stays open until someone closes it.
+
+You create a <xref:System.Threading.EventWaitHandle> instance that represents a named system synchronization event by using one of the [EventWaitHandle constructors](<xref:System.Threading.EventWaitHandle.%23ctor%2A>) that specifies a name or the <xref:System.Threading.EventWaitHandle.OpenExisting%2A?displayProperty=nameWithType> method. A named system synchronization event can be used for the inter-process synchronization.
+
+For more information, see the [EventWaitHandle](eventwaithandle.md) article and the <xref:System.Threading.EventWaitHandle> API reference.
+
+### AutoResetEvent class
+
+The <xref:System.Threading.AutoResetEvent?displayProperty=nameWithType> class derives from <xref:System.Threading.EventWaitHandle> and represents a synchronization event that, when signaled, resets automatically to an unsignaled state after releasing a single waiting thread.
+
+The <xref:System.Threading.AutoResetEvent> class cannot represent a named system synchronization event and cannot be used for the inter-process synchronization.
```

Comment:
Note that you can have a named auto/manual reset event if you use the EventWaitHandle class directly. That's one more reason to avoid having these 2 separate sections.